src/main/java/org/motechproject/newebodac/service/ImportService.java

```diff
@@ -0,0 +1,266 @@
+            continue;
+          }
+          visit.setVaccinee(correspondingVaccinee);
+
+          Visit existingVisit = visitRepository
+              .getByVaccineeAndType(visit.getVaccinee(), visit.getType());
+          if (existingVisit != null) {
+            visit.setId(existingVisit.getId());
+          }
+          visitRepository.save(visit);
+          break;
+        case PERSON:
+          handlePersonEntity(cleanRow, mapper);
+          break;
+        default:
+          break;
+      }
+    }
+    csvMapReader.close();
+    return errorMap;
+  }
+
+  private String[] getCsvHeaders(ICsvMapReader csvMapReader) throws IOException {
+    return Arrays.stream(csvMapReader.getHeader(true))
+          .map(String::trim)
+          .map(String::toLowerCase)
+          .toArray(String[]::new);
+  }
+
+  private List<String> getAllCsvConfigFieldNames(CsvConfig csvConfig) {
+    return csvConfig.getCsvFields()
+        .stream()
+        .map(CsvField::getFieldName)
+        .map(String::toLowerCase)
+        .collect(Collectors.toList());
+  }
+
+  private void configureMapper() {
+    mapper.registerModule(new JavaTimeModule());
+    mapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+  }
+
+  private void handlePersonEntity(Map<String, Object> cleanRow, ObjectMapper mapper) {
+    KeyCommunityPerson keyCommunityPerson = mapper.convertValue(cleanRow,
+        KeyCommunityPerson.class);
+
+    keyCommunityPersonRepository.save(keyCommunityPerson);
+
+    Language correspondingLanguagee = languageRepository
+        .getOne(keyCommunityPerson.getLanguage().getId());
+    if (correspondingLanguagee != null) {
+      keyCommunityPerson.setLanguage(correspondingLanguagee);
+    }
+  }
+
+  private void handleVaccineeEntity(Map<String, Object> cleanRow, ObjectMapper mapper) {
+    Vaccinee vaccinee = mapper.convertValue(cleanRow, Vaccinee.class);
+
+    Language correspondingLanguage = languageRepository
+        .getOne(vaccinee.getPreferredLanguage().getId());
+    if (correspondingLanguage != null) {
+      vaccinee.setPreferredLanguage(correspondingLanguage);
+    }
+
+    Vaccinee existingVaccinee = vaccineeRepository.findByVaccineeId(vaccinee.getVaccineeId());
+    if (existingVaccinee != null) {
+      vaccinee.setId(existingVaccinee.getId());
+    }
+    vaccineeRepository.save(vaccinee);
+  }
+
+  private void addFieldValueMapErrors(ICsvMapReader csvMapReader, Map<Integer, String> errorMap,
+      String cellValue, CsvField csvField) {
+    List<String> mappedValues = new ArrayList<>();
+    csvField.getFieldValueMap().forEach((k, v) -> mappedValues.add(k));
+    errorMap.put(csvMapReader.getLineNumber(), String.format(
+        "The value '%s' from the CSV isn't mapped in the CSV values config: "
+            + "%s", cellValue, mappedValues));
+  }
+
+  private Class getRelatedEntityClass(CsvField csvField) throws ClassNotFoundException {
+    return Class.forName(DOMAIN_PACKAGE_PREFIX
```

Comment:
you could just write some helper for this with map, key will be the enum and value will be the class